### PR TITLE
feat: Upload support `maxCount`

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Upload support `maxCount` to limit file count.      |
| 🇨🇳 Chinese |     Upload 支持 `maxCount` 属性以限制文件数量。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/upload/demo/max-count.md](https://github.com/ant-design/ant-design/blob/upload-max-count/components/upload/demo/max-count.md)

### CODE REVIEW

File: components/upload/demo/max-count.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+order: 10
+title:
+  zh-CN: 限制数量
+  en-US: Max Count
+---
+
+## zh-CN
+
+通过 `maxCount` 限制上传数量。当为 `1` 时，始终用最新上传的代替当前。
```
Comment (afc163) [inline components/upload/demo/max-count.md:10]: 默认模式是替代还是禁止，感觉禁止比较好，类似 Select tags 或者 Checkbox 的最大限制，肯定不希望选了一个新的之后把旧的顶出去。

Comment (zombieJ) [reply components/upload/demo/max-count.md:10]: 云凤蝶那边问下来单个的文件没有禁止的需求，只有自动替换的需求。所以这边给 `maxCount=1` 做了一个特殊的规则。

Comment (afc163) [reply components/upload/demo/max-count.md:10]: 先上吧，看看社区反馈。

**Comment by afc163** [comment]
feature 掉了一行测试覆盖。

**Review by afc163 — APPROVED** [review]

**Comment by zombieJ** [comment]
modal 的 coverage 掉了和这个无关，外面发新 PR 修好了。

**Pull request merged**